integration-tests/gatsby-pipeline/__tests__/lazy-image-build/develop.js

```diff
@@ -0,0 +1,162 @@
+const { execFileSync } = require(`child_process`)
+const execa = require(`execa`)
+const path = require(`path`)
+const fs = require(`fs-extra`)
+const basePath = path.resolve(__dirname, `../../`)
+const request = require(`request-promise-native`)
+
+// 2 min
+jest.setTimeout(2000 * 60)
+
+const bootDevelop = () =>
+  new Promise(resolve => {
+    const devProcess = execa(`yarn`, [`develop`], {
+      cwd: basePath,
+      env: { NODE_ENV: `development` },
+      detached: true,
+    })
+
+    devProcess.stdout.on(`data`, chunk => {
+      if (chunk.toString().includes(`You can now view`)) {
+        // resolving the devProcess wans't seem to resolve the promise so it's wrapped in an object.
+        resolve({ process: devProcess })
+      }
+    })
+  })
+
+const killDevelop = devProcess =>
+  new Promise(resolve => {
+    if (!devProcess) {
+      return resolve()
+    }
+
+    devProcess.on(`exit`, code => {
+      // give it some time to exit
+      setTimeout(() => {
+        resolve()
+      }, 0)
+    })
+
+    return process.kill(-devProcess.pid)
```

Comment:
For my edificaition--what is this doing exactly? Does this need to be negative?

Reply:
i'll add a comment to it with the following link https://stackoverflow.com/a/33367711
it actually kills every process in that group